testframework/src/main/java/net/neoforged/testframework/impl/GameTestHelperFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.testframework.impl;
+
+import com.mojang.logging.LogUtils;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodType;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import net.minecraft.Util;
+import net.minecraft.gametest.framework.GameTestHelper;
+import net.minecraft.gametest.framework.GameTestInfo;
+import net.neoforged.testframework.gametest.ExtendedGameTestHelper;
+
+public interface GameTestHelperFactory<T extends GameTestHelper> extends Function<GameTestInfo, T> {
+    Map<Class<?>, GameTestHelperFactory<?>> CONSTRUCTORS = Util.make(new HashMap<>(), map -> {
```

Comment:
Can be replaced with an identityhashmap and it would be faster.

Reply:
Done